feat: new rule to detect enum value that do not respect specified type (#913)
stoplightio/spectral: docs/reference/functions.md

      - externalValue
      - value
```

## typedEnum

> Models described by both a `type` and an `enum` must only expose enum values that respect this type.

Reviewer: ```suggestion
When both a `type` and `enum` are defined for a property, the enum values must respect the type.
```

This makes me think this would be better off as a custom function for the oas ruleset. Afterall, this is oas/json schema specific, which means it doesn't really belong in core functions, right?

<!-- title: functionOptions -->

name                   | description                                                        | required?
-----------------------|--------------------------------------------------------------------|----------

> reportingThreshold     | the maximum number of invalid values reported in the error message | yes

Reviewer: This seems unique to this rule, whats the idea here?

      - externalValue
      - value
```

> <!-- 

Author: @philsturgeon Switching to custom function means I can drop that, as we only document core functions, right?

Reviewer: Correct, we document the OpenAPI rules and the function is just an implementation detail.
